x-pack/plugins/cases/public/components/all_cases/assignees_filter.tsx

```diff
@@ -42,9 +52,17 @@ const AssigneesFilterPopoverComponent: React.FC<AssigneesFilterPopoverProps> = (
   const togglePopover = useCallback(() => setIsPopoverOpen((value) => !value), []);
 
   const onChange = useCallback(
-    (users: UserProfileWithAvatar[]) => {
-      const sortedUsers = bringCurrentUserToFrontAndSort(currentUserProfile, users);
-      onSelectionChange(sortedUsers ?? []);
+    (users: AssigneesFilteringSelection[]) => {
+      const usersWithNoAssigneeSelection = removeNoAssigneesSelection(users);
+      const sortedUsers =
+        bringCurrentUserToFrontAndSort(currentUserProfile, usersWithNoAssigneeSelection) ?? [];
```

Comment:
Would it make the code easier if we modified `bringCurrentUserToFrontAndSort` and related functions to handle `null`? Maybe they could put the nulls at the front and we could just rename it to `orderAssignees` or something like that.